### PR TITLE
Put ZSS in an independent launch group (backward-compatible)

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
This allows for ZSS to be started without the desktop by specifying LAUNCH_COMPONENT_GROUPS=ZSS
The default of `LAUNCH_COMPONENT_GROUPS=DESKTOP,GATEWAY` will still work, as ZSS is added if DESKTOP is requested and ZSS is not seen in the list.

Why do this? Primarily due to Docker, we want to make it easy to turn on just ZSS in that case.

How to test? If ZSS started twice, there would be 2 zssServer log files in the logs folder. It should only start once even if requested by both GROUP ZSS & DESKTOP.

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 

- [x] Feature 

#### Changes proposed in this PR
The relationship between ZSS and app-server were not fully exposed previously. In order to use ZSS, you had to also have app-server running. But, this is not really how it works. I'm moving ZSS into its own component group so that it can be started independently, but only if people specify that. If they dont, and just continue to specify DESKTOP, then zss should still load as a prereq to app-server. Additionally, zss is not present on Docker, so I do not put it into the list if this script is not running on z/os.

#### Does this PR introduce a breaking change?

- [x] No

The documentation should be amended to explain this option to have ZSS without app-server, although I don't think it will be necessary for most all-zos installs, it will be useful for docker users.

### CODE REVIEW

**Review by stevenhorsman — CHANGES_REQUESTED** [review]
Also need test runs linked

**Review by jackjia-ibm — COMMENTED** [review]
I'm not 100% sure. Will this have any compatibility issue if user upgrade from FMID? In that case, the user won't have ZSS in LAUNCH_COMPONENT_GROUPS defined.

**Comment by Joe-Winchester** [comment]
> I'm not 100% sure. Will this have any compatibility issue if user upgrade from FMID? In that case, the user won't have ZSS in LAUNCH_COMPONENT_GROUPS defined.

The component groups are in each instance.env file, so the absence of ZSS should be OK because a user that has DESKTOP will get ZSS launched following Sean's logic.  If a user adds ZSS and DESKTOP this is belt and braces.  The instance.env is in the instance directory which is not affected by an SMP/E (or convenience) upgrade.  We could add ZSS as a new default so that when zowe-configure-instance.sh is run the three values are DESKTOP, ZSS, GATEWAY which would let folks know they can just have ZSS by itself which might be a nice touch, however ZSS only is going to just be for the docker container scenario in which case a lot of the other instance.env variables are not required (Java, Node, lots of ports) so maybe there should be a `zowe-configure-instance.sh` arg that creates a stripped down instance.env and directory that is just optimized for supporting containers only, and also has some steps to help with AT-TLS.

File: bin/internal/run-zowe.sh [hunk 1]
```diff
@@ -97,10 +97,27 @@ then
...
+    LAUNCH_COMPONENTS=zss,${LAUNCH_COMPONENTS} #zss should run before app-server to reduce the polling period when app-server tries to find it.
+  fi
+fi
+
 #Explorers may be present, but have a prereq on gateway, not desktop
 if [[ $LAUNCH_COMPONENT_GROUPS == *"DESKTOP"* ]]
 then
-  LAUNCH_COMPONENTS=zss,app-server,${LAUNCH_COMPONENTS} #Make app-server the first component, so any extender plugins can use its config
+  #In each below, put app-server before other components, so any extender plugins can use its config
+  if [ `uname` = "OS/390" ]; then
```
[the remark this answers is not in the document]
Comment (1000TurquoisePogs) [reply bin/internal/run-zowe.sh:111]: i will see if i can do that in a way that is cross-platform

**Comment by 1000TurquoisePogs** [comment]
>so maybe there should be a zowe-configure-instance.sh arg that creates a stripped down instance.env and directory that is just optimized for supporting containers only, and also has some steps to help with AT-TLS.
Very possibly Joe. I'm reviewing all the documentation and it's a huge task to compartmentalize it so that the users are not surprised when they dont see missing variables.
I think what would be good is in the case that docker is intended, the instance.env has all the unneeded variables as commented out with a description about why. Then, they can uncomment them if they want to.

**Comment by 1000TurquoisePogs** [comment]
I will work on Steve's test & refactoring comment.

**Pull request closed without merging**